client/state/plugins/recommended/reducer.js

```diff
@@ -0,0 +1,19 @@
+/** @format */
+/**
+ * Internal dependencies
+ */
+import { PLUGINS_RECOMMENDED_REQUEST, PLUGINS_RECOMMENDED_RECEIVE } from 'state/action-types';
+
+export const items = ( state = {}, action ) => {
+	switch ( action.type ) {
+		case PLUGINS_RECOMMENDED_REQUEST: {
+			return { ...state, [ action.siteId ]: null };
+		}
+		case PLUGINS_RECOMMENDED_RECEIVE: {
+			return { ...state, [ action.siteId ]: action.data };
+		}
+	}
+	return state;
+};
```

Comment:
You could use `keyedReducer` here:
```js
export default keyedReducer( 'siteId', ( state = null, action ) => {
  switch ( action.type ) {
    case REQUEST:
      return null; // not even needed
    case RECEIVE:
      return action.data;
    default:
      return state;
  }
} );
```
We usually don't even set the state to `null` when the request is in flight -- why would we? If there is already data in state from previous request and we navigate away from and back to the route, we can show the existing data, not a placeholder, and refresh the data in background.

Reply:
Reflecting on your comment made me realize what I actually wanted was a way to handle network request failures. The reason you laid out for using `keyedReducer` and getting rid of `null` makes a lot of sense 🙇